x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/render_table_columns/render_endpoint/render_endpoint.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { InferenceAPIConfigResponse } from '@kbn/ml-trained-models-utils';
+import { RenderElasticsearch } from './render_elasticsearch';
+import { RenderCohere } from './render_cohere';
+import { ServiceProviderKeys } from '../../types';
+import { RenderHuggingFace } from './render_hugging_face';
+import { RenderOpenAI } from './render_open_ai';
+import { RenderAzureAIStudio } from './render_azure_ai_studio';
+import { RenderMistral } from './render_mistral';
+import { RenderAzureOpenAI } from './render_azure_open_ai';
+import { RenderGoogleAIStudio } from './render_google_ai_studio';
+
+export interface RenderEndpointProps {
+  endpoint: InferenceAPIConfigResponse;
+}
+
+type RenderMapType = {
+  [key in ServiceProviderKeys]?: JSX.Element;
+};
+
+export const RenderEndpoint: React.FC<RenderEndpointProps> = ({ endpoint }) => {
+  const renderMap: RenderMapType = {
+    [ServiceProviderKeys.elser]: <RenderElasticsearch endpoint={endpoint} />,
```

Comment:
All of these components seem very similar with small differences. Seems like abstracting a generic component with props might have been a better approach here vs implementing 9 very similar components 🤷‍♂️

Reply:
Yup, I agree. We should have one Endpoint component that can handle all of these types.

Reply:
Moved all the related components to a single component